include/alpaka/acc/AccGenericSycl.hpp

```diff
@@ -0,0 +1,186 @@
+#    include <alpaka/intrinsic/IntrinsicGenericSycl.hpp>
+#    include <alpaka/math/MathGenericSycl.hpp>
+#    include <alpaka/mem/fence/MemFenceGenericSycl.hpp>
+#    include <alpaka/warp/WarpGenericSycl.hpp>
+#    include <alpaka/workdiv/WorkDivGenericSycl.hpp>
+
+// Specialized traits.
+#    include <alpaka/acc/Traits.hpp>
+#    include <alpaka/dev/Traits.hpp>
+#    include <alpaka/idx/Traits.hpp>
+#    include <alpaka/kernel/Traits.hpp>
+#    include <alpaka/pltf/Traits.hpp>
+
+// Implementation details.
+#    include <alpaka/core/BoostPredef.hpp>
+#    include <alpaka/core/ClipCast.hpp>
+#    include <alpaka/core/Sycl.hpp>
+
+#    include <CL/sycl.hpp>
+
+#    include <cstddef>
+#    include <string>
+#    include <type_traits>
+
+namespace alpaka
+{
+    namespace experimental
+    {
+        //! The SYCL accelerator.
+        //!
+        //! This accelerator allows parallel kernel execution on SYCL devices.
+        template<typename TDim, typename TIdx>
+        class AccGenericSycl
+            : public WorkDivGenericSycl<TDim, TIdx>
+            , public gb::IdxGbGenericSycl<TDim, TIdx>
+            , public bt::IdxBtGenericSycl<TDim, TIdx>
+            , public AtomicHierarchy<AtomicGenericSycl, AtomicGenericSycl, AtomicGenericSycl>
+            , public math::MathGenericSycl
+            , public BlockSharedMemDynGenericSycl
+            , public BlockSharedMemStGenericSycl
+            , public BlockSyncGenericSycl<TDim>
+            , public IntrinsicGenericSycl
+            , public MemFenceGenericSycl
+            , public warp::WarpGenericSycl<TDim>
+        {
+        public:
+#    ifdef ALPAKA_SYCL_STREAM_ENABLED
+            AccGenericSycl(
+                Vec<TDim, TIdx> const& threadElemExtent,
+                sycl::nd_item<TDim::value> work_item,
+                sycl::accessor<std::byte, 1, sycl::access_mode::read_write, sycl::target::local> shared_acc,
+                sycl::accessor<int, 1, sycl::access_mode::read_write, sycl::target::global_buffer> global_fence_dummy,
+                sycl::accessor<int, 1, sycl::access_mode::read_write, sycl::target::local> local_fence_dummy,
+                sycl::stream output_stream)
+                : WorkDivGenericSycl<TDim, TIdx>{threadElemExtent, work_item}
+                , gb::IdxGbGenericSycl<TDim, TIdx>{work_item}
+                , bt::IdxBtGenericSycl<TDim, TIdx>{work_item}
+                , AtomicHierarchy<AtomicGenericSycl, AtomicGenericSycl, AtomicGenericSycl>{}
+                , math::MathGenericSycl{}
+                , BlockSharedMemDynGenericSycl{shared_acc}
+                , BlockSharedMemStGenericSycl{}
+                , BlockSyncGenericSycl<TDim>{work_item}
+                , IntrinsicGenericSycl{}
+                , MemFenceGenericSycl{global_fence_dummy, local_fence_dummy}
+                , warp::WarpGenericSycl<TDim>{work_item}
+                , stream{output_stream}
+            {
+            }
+
+            sycl::stream stream;
```

Comment:
I cannot find where this is used.

Reply:
In the SYCL implementation of `assert` IIRC.

Reply:
It is used in `include/alpaka/test/Check.hpp`.